src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -170,4 +171,47 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String balance} into a {@code Balance}.
+     * Handles the conversion of a dollar amount into cents.
+     */
+    public static Balance parseBalance(String balance) throws ParseException {
+        requireNonNull(balance);
+        String trimmedBalance = balance.trim();
+
+
+        if (!Balance.isValidDollarString(trimmedBalance)) {
+            throw new ParseException(Balance.MESSAGE_CONSTRAINTS);
+        }
+
+        // Remove the dollar sign if it exists
+        String dollarAmount = trimmedBalance.replace("$", "");
+
+        // Split the string at the decimal point
+        String[] parts = dollarAmount.split("\\.");
+
+        // Check that the dollar amount does not clearly exceed 5 digits.
+        // This prevents integer overflow when converting to cents and when
+        // performing validation in subsequent pay / owe operations.
+        if (parts[0].length() > 5) {
```

Comment:
Very lame comment but:
$000000000.00 fails (even though actual dollar amount is $0.00).

Reply:
Yo good test case actually, I'll strip leading 0s

Reply:
Fix + corresponding tests in https://github.com/AY2324S1-CS2103T-T13-2/tp/pull/89/commits/b96845d12a7567c1b6d953675075e1951c84ec4b, thanks again!